yaml-tests/src/test/java/MultiServerConnectionFactoryTest.java

```diff
@@ -42,61 +43,119 @@ void testDefaultPolicy(int initialConnection) throws SQLException {
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
+
         connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(0, connection.getCurrentConnectionSelector());
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, 0, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
+
         connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(0, connection.getCurrentConnectionSelector());
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, 0, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
     }
 
     @ParameterizedTest
     @CsvSource({"0", "1"})
     void testAlternatePolicy(int initialConnection) throws SQLException {
+        final String[] path = new String[] { "Primary", "Alternate" };
+
         MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
                 MultiServerConnectionFactory.ConnectionSelectionPolicy.ALTERNATE,
                 initialConnection,
-                dummyConnectionFactory(),
-                List.of(dummyConnectionFactory()));
+                dummyConnectionFactory("Primary"),
+                List.of(dummyConnectionFactory("Alternate")));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
-        MultiServerConnectionFactory.MultiServerRelationalConnection connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(initialConnection % 2, connection.getCurrentConnectionSelector());
+
+        // First run:
+        // - Factory current connection: initial connection
+        // - connection current connection: initial connection
+        // - statement: initial connection
+        MultiServerConnectionFactory.MultiServerRelationalConnection connection =
+                (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
         assertEquals((initialConnection + 1) % 2, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, initialConnection, path[initialConnection]);
+        RelationalPreparedStatement statement = connection.prepareStatement("SQL");
+        assertStatement(statement, path[initialConnection]);
```

Comment:
I feel like it would be clearer to have the names flip above, so that all of these can just be
```
assertStatement(connection.prepareStatement("SQL"), "Initial");
assertStatement(connection.prepareStatement("SQL"), "Other");
assertStatement(connection.prepareStatement("SQL"), "Initial");
```

Reply:
Done (I think what you meant was to assign names to the `path[initial]` and `path[initial+1]` values)